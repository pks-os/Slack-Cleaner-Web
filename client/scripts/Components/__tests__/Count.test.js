import React from 'react';
import renderer from 'react-test-renderer';

import Count from '../Count';

const MOCK_DATA = [{ size: 300 }, { size: 200 }, { size: 500 }];

const MOCK_DATA_SMALL = [{ size: 300 }];

describe('<Checkbox />', () => {
  it('renders correctly', () => {
    const count = renderer.create(<Count />).toJSON();
    expect(count).toMatchSnapshot();
  });

  it('displays the correct data', () => {
    const count = renderer
      .create(<Count data={MOCK_DATA} total={MOCK_DATA.length} />)
      .toJSON();
    expect(count).toMatchSnapshot();
  });

  it('displays the correct data when a smaller total is passed', () => {
    const count = renderer
      .create(<Count data={MOCK_DATA_SMALL} total={MOCK_DATA_SMALL.length} />)
      .toJSON();
    expect(count).toMatchSnapshot();
  });
});
