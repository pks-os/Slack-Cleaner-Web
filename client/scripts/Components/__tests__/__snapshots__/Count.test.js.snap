// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> displays the correct data 1`] = `
<div
  className="Count"
>
  <p
    className="Count__Text"
  >
    There are
     
    <span
      className="purple"
    >
      3
       
      files
    </span>
     
    you can delete from your workspace.
  </p>
  <p
    className="Count__Text"
  >
    Here are
     
    3
     worth
     
    <span
      className="red"
    >
      1000 Bytes
    </span>
  </p>
</div>
`;

exports[`<Checkbox /> displays the correct data when a smaller total is passed 1`] = `
<div
  className="Count"
>
  <p
    className="Count__Text"
  >
    There is
     
    <span
      className="purple"
    >
      1
       
      file
    </span>
     
    you can delete from your workspace.
  </p>
  <p
    className="Count__Text"
  >
    Here is
     
    1
     worth
     
    <span
      className="red"
    >
      300 Bytes
    </span>
  </p>
</div>
`;

exports[`<Checkbox /> renders correctly 1`] = `
<div
  className="Count"
>
  <p
    className="Count__Text"
  >
    There is
     
    <span
      className="purple"
    >
       
      file
    </span>
     
    you can delete from your workspace.
  </p>
  <p
    className="Count__Text"
  >
    Here is
     
    0
     worth
     
    <span
      className="red"
    >
      0 Bytes
    </span>
  </p>
</div>
`;
